src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/GetStaffDetails.kt

```diff
@@ -51,7 +52,28 @@ class GetStaffDetails(
   private val caseNotesApiClient: CaseNotesApiClient,
   private val referenceDataRepository: ReferenceDataRepository,
 ) {
-  fun getFor(
+  fun getJobClassificationsFor(
+    prisonCode: String,
+    staffId: Long,
+  ): JobClassificationResponse {
+    val policies: Set<AllocationPolicy> =
+      buildSet {
+        if (prisonApi.getKeyworkerForPrison(prisonCode, staffId) != null) add(AllocationPolicy.KEY_WORKER)
```

Comment:
I think this should be calling staffRoleCheck rather than getKeyworkerForPrison. staffRoleCheck uses the staff specific /staff/{staffId}/{prisonCode}/roles/{role} endpoint whereas getKeyworkerForPrison retrieves all the KWs